src/rpc/rawtransaction.cpp

```diff
@@ -1791,6 +1792,320 @@ UniValue converttopsbt(const JSONRPCRequest& request)
+            continue;
+        }
+
+        const Coin& coin = view.AccessCoin(psbtx.tx->vin[i].prevout);
+
+        std::vector<std::vector<unsigned char>> solutions_data;
+        txnouttype which_type = Solver(coin.out.scriptPubKey, solutions_data);
+        if (which_type == TX_WITNESS_V0_SCRIPTHASH || which_type == TX_WITNESS_V0_KEYHASH || which_type == TX_WITNESS_UNKNOWN) {
+            input.witness_utxo = coin.out;
+        }
+    }
+
+    CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
+    ssTx << psbtx;
+    return EncodeBase64((unsigned char*)ssTx.data(), ssTx.size());
+}
+
+UniValue joinpsbts(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+                            "joinpsbts [\"psbt\",...]\n"
+                            "\nJoins multiple distinct PSBTs with different inputs and outputs into one PSBT with inputs and outputs from all of the PSBTs\n"
+                            "No input in any of the PSBTs can be in more than one of the PSBTs.\n"
+                            "\nArguments:\n"
+                            "1. \"txs\"                   (string) A json array of base64 strings of partially signed transactions\n"
+                            "    [\n"
+                            "      \"psbt\"             (string) A base64 string of a PSBT\n"
+                            "      ,...\n"
+                            "    ]\n"
+
+                            "\nResult:\n"
+                            "  \"psbt\"          (string) The base64-encoded partially signed transaction\n"
+                            "\nExamples:\n"
+                            + HelpExampleCli("joinpsbts", "\"psbt\"")
+                            );
+
+    RPCTypeCheck(request.params, {UniValue::VARR}, true);
+
+    // Unserialize the transactions
+    std::vector<PartiallySignedTransaction> psbtxs;
+    UniValue txs = request.params[0].get_array();
+    for (unsigned int i = 0; i < txs.size(); ++i) {
+        PartiallySignedTransaction psbtx;
+        std::string error;
+        if (!DecodePSBT(psbtx, txs[i].get_str(), error)) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+        }
+        psbtxs.push_back(psbtx);
+    }
+
+    PartiallySignedTransaction merged_psbt(psbtxs[0]); // Copy the first one
+
+    // Clear signatures from all inputs
+    for (auto& input : merged_psbt.inputs) {
+        input.partial_sigs.clear();
+    }
+
+    // Merge
+    for (auto it = std::next(psbtxs.begin()); it != psbtxs.end(); ++it) {
+        for (unsigned int i = 0; i < it->tx->vin.size(); ++i) {
+            if (!merged_psbt.AddInput(it->tx->vin[i], it->inputs[i])) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "An input exists in multiple PSBTs");
+            }
+        }
+        for (unsigned int i = 0; i < it->tx->vout.size(); ++i) {
+            merged_psbt.AddOutput(it->tx->vout[i], it->outputs[i]);
```

Comment:
Should allow duplicate outputs? Or should sum values into one output?

Reply:
@achow101 could you ack/nack on #12419, esp @MeshCollider https://github.com/bitcoin/bitcoin/pull/12419#issuecomment-437636604